File: app/src/main/java/ru/vpiska/helper/DatePicker.java
package ru.vpiska.helper;


import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;

import android.widget.Button;
import android.widget.TextView;

import java.util.Calendar;

import ru.vpiska.R;

public class DatePicker extends DialogFragment
        implements DatePickerDialog.OnDateSetListener {



    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // определяем текущую дату


        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR)-18;
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);


        // создаем DatePickerDialog и возвращаем его
        DatePickerDialog picker = new DatePickerDialog(getActivity(),AlertDialog.THEME_HOLO_LIGHT, this,
                year, month, day);
        picker.setTitle(getResources().getString(R.string.choose_date));

        long now = System.currentTimeMillis() - 1000;
        c.set(year-99, month, day);

        picker.getDatePicker().setMinDate(c.getTimeInMillis());
        picker.getDatePicker().setMaxDate(now);

        return picker;
    }

    @Override
    public void onStart() {
        super.onStart();
        // добавляем кастомный текст для кнопки
        Button nButton = ((AlertDialog) getDialog())
                .getButton(DialogInterface.BUTTON_POSITIVE);
        nButton.setText(getResources().getString(R.string.ready));
        Button ngButton = ((AlertDialog) getDialog())
                .getButton(DialogInterface.BUTTON_NEGATIVE);
        ngButton.setText(getResources().getString(R.string.cancel));

    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onDateSet(android.widget.DatePicker datePicker, int year,
                          int month, int day) {

        TextView txtAge = (TextView) getActivity().findViewById(R.id.txtAge);
        month += (int)1;
        txtAge.setText(day + "-" + month + "-" + year);
    }
}

File: app/src/main/java/ru/vpiska/helper/DatePicker.kt
package ru.vpiska.helper


import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.DatePickerDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.view.View
import android.widget.TextView
import ru.vpiska.R
import java.util.*

class DatePicker : DialogFragment(), DatePickerDialog.OnDateSetListener {


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // определяем текущую дату


        val c = Calendar.getInstance()
        val year = c.get(Calendar.YEAR) - 18
        val month = c.get(Calendar.MONTH)
        val day = c.get(Calendar.DAY_OF_MONTH)


        // создаем DatePickerDialog и возвращаем его
        val picker = DatePickerDialog(activity!!, AlertDialog.THEME_HOLO_LIGHT, this,
                year, month, day)
        picker.setTitle(resources.getString(R.string.choose_date))

        val now = System.currentTimeMillis() - 1000
        c.set(year - 99, month, day)

        picker.datePicker.minDate = c.timeInMillis
        picker.datePicker.maxDate = now

        return picker
    }

    override fun onStart() {
        super.onStart()
        // добавляем кастомный текст для кнопки
        val nButton = (dialog as AlertDialog)
                .getButton(DialogInterface.BUTTON_POSITIVE)
        nButton.text = resources.getString(R.string.ready)
        val ngButton = (dialog as AlertDialog)
                .getButton(DialogInterface.BUTTON_NEGATIVE)
        ngButton.text = resources.getString(R.string.cancel)

    }

    @SuppressLint("SetTextI18n")
    override fun onDateSet(datePicker: android.widget.DatePicker, year: Int,
                           month: Int, day: Int) {
        var month = month

        val txtAge = activity!!.findViewById<View>(R.id.txtAge) as TextView
        month += 1
        txtAge.text = "$day-$month-$year"
    }
}